nemo/collections/nlp/data/dialogue/data_processor/data_processor.py

```diff
@@ -50,3 +53,29 @@ def get_dev_examples(self):
     def get_test_examples(self):
         """Gets a collection of `InputExample`s for the test set."""
         raise NotImplementedError()
+
+    @staticmethod
+    def get_relevant_idxs(dataset_split, n_samples, dev_proportion):
+        """
+        Obtain indexes for each dataset_split, when train and dev sets are not in separate files
+        
+        Args: 
+            dataset_split: train, dev or test
+            n_samples: total number of samples
+            dev_proportion: value from 1 to 99 that represent proportion of data in dev set
+        """
+
+        if dataset_split in ["train", "dev"]:
+            n_dev = int(n_samples * (dev_proportion / 100))
+            dev_idxs = random.sample(list(range(n_samples)), n_dev)
+            if dataset_split == "dev":
+                idxs = dev_idxs
+            else:
+                dev_idxs_set = set(dev_idxs)
+                train_idxs = [idx for idx in list(range(n_samples)) if idx not in dev_idxs_set]
+                idxs = train_idxs
+
+        elif dataset_split == "test":
```

Comment:
please make sure every
```
if ...
elif ...
else
``` 
block ends with ``else``. In this specific case, in else you'd raise an exception and tell the user supported names of splits (train, dev, test)

Reply:
sure